rgo/git.py

```diff
@@ -216,19 +216,6 @@ def describe(self, pkg=None, spec_version=None, version_from="git"):
             version = "0"
             release = "0.{!s}.{!s}.g{!s}".format(date, commits, hash)
 
-        elif commits == 0:
-            # release
-
-            # we'll set version to git_version
-            # it's a release, we'll set release to 1
-
-            # Example:
-            #    spec:   doesn't matter
-            #    git:    version=1.9; there are no commits after the tag
-            #    result: version=0, release=0.YYYYMMDDhhmmss.3.gcafecafe
-            version = git_version
-            release = "1"
```

Comment:
I'm not sure about this. I'm not necessarily opposed, but I think if we want to always install packages built for the test, maybe we should force their installation without a regard for the version, instead of making their version always be higher.

Regarding this actual change in general, the downstream patches throw a spanner in the works no matter what. The reason is we don't know if the downstream patches e.g. revert an upstream change (virtually bringing the version "lower", though it breaks the linearity in the sense that the reverted change can be several versions old), or adding a change that could semantically be considered a new feature that could be considered a version "increase". Or the change could be completely orthogonal.

Disregarding downstream patches it makes sense to make the version be the same, as we expect the package content to be the same...

But maybe your change makes sense because it makes the built packages versions always higher, so in a nightly repo scenario it would ensure the packages from such repo would be installed more consistently, not ambiguously jumping between distro repo and the nightly in case the versions end up being equal (which can cause unexpected and non-deterministic behavior in case of downstream patches).

@dmach ping in case you have opinions :slightly_smiling_face:

Reply:
> I'm not sure about this. I'm not necessarily opposed, but I think if we want to always install packages built for the test, maybe we should force their installation without a regard for the version, instead of making their version always be higher.

That does sound reasonable, I made a PR: https://github.com/rpm-software-management/ci-dnf-stack/pull/1080, we can use that regardless of what we decide here.

> Disregarding downstream patches it makes sense to make the version be the same, as we expect the package content to be the same...

Yes the version is the same in both cases, but to me it seems reasonable that the release part of nevra is different. The nightly is a different release from the `-1.fc35` fedora release. The way I see it the mentioned downstream patches for example make it a different release.

From a practical stand point I also like that the naming is consistent and I am able to easily tell which package is from nightly and which from fedora.